google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/MavenBuildDeploymentSourceType.java

```diff
@@ -107,6 +110,11 @@ public void save(@NotNull ModuleDeploymentSource deploymentSource, @NotNull Elem
       if (deployable.getVersion() != null) {
         tag.setAttribute(VERSION_ATTRIBUTE, ((AppEngineDeployable) deploymentSource).getVersion());
       }
+
+      if (deployable.getEnvironment() != null) {
```

Comment:
if the environment happens to be null, won't the deserialization (load()) method throw an exception when doing valueOf?